### ISSUE TITLE
Override Global Function Aliases used in the Core?

### ISSUE THREAD

**Issue by lukearmstrong** [fuel/core#698]
Fuel defines several functions in the global namespace to serve as aliases to methods in Core Classes.

For example `__()` returns `\Lang::get()`

But I wanted to define my own `__()` function, though I can't do that safely unless I grep the core and replace all usages of the `__()` alias with `\Lang::get()`. For obvious reasons I don't want to modify any files in the core, so I won't be doing that.

To me, an alias is something that should be defined for developers to use, not for the core to use - as that will avoid the problem I am having now. So either I pick another function name to use, or I figure out a way of getting the `Lang` class to use `gettext()`.

Also, `/fuel/core/base.php` checks if the function exists. Is it possible to define my function before this is run?

**Comment by jschreuder** [comment]
As far as I know `__()` isn't used anywhere in the core, thus that's actually the worst example you could have chosen. Define any `__()` function before it (in the APP bootstrap for example) and you won't have any trouble. Same goes for `e()` and `render()`.

All the other methods in `base.php` are used by the core and can't be savely overwritten. They're mostly there for optimization  (ex: `logger()`) or to augment what PHP is lacking (ex `in_arrayi()`). This won't change for the 1.x branch but are likely to be removed in 2.0.

**Comment by lukearmstrong** [comment]
Not quite sure what your problem is... but the `__()` alias is used in the core, and in your ORM package.

```
./includes/fuel/core/classes/validation/error.php:              ? __('validation.'.$this->rule) ?: __('validation.'.Arr::get(explode(':', $this->rule), 0))
./includes/fuel/core/classes/form.php:      return html_tag('label', $attributes, __($label) ?: $label);
./includes/fuel/packages/orm/classes/observer/validation.php:                   $settings['form']['options'][$key] = __($value) ?: $value;
```

**Comment by jschreuder** [comment]
> Not quite sure what your problem is... but the __() alias is used in the core, and in your ORM package.

Ah, so it is used, I shouldn't have done that. I'm very sorry my memory isn't perfect - wait I'm not, I don't give a rats ass. Help out, be constructive and notice when I use wording like the above that leave room to be contradicted.

**Comment by lukearmstrong** [comment]
I'm still completely oblivious as to why you are taking this tone with me. My previous comment wasn't supposed to be taken personally, it was to clarify the cases where it is used in the hope that this issue would be reopened.

If I submitted a pull request to `fuel/core` and `fuel/orm` where usages of `__()` were replaced with `\Lang::get()` would it be accepted?

That way I can do as you advise, and define my `__()` function before `require COREPATH.'bootstrap.php';` in `fuel/app/bootstrap.php` and still have clean Core and ORM folders that can be updated.
